p2p/node/p2p_services.go

```diff
@@ -77,9 +78,22 @@ func (p *P2PNode) requestFromPeer(peerID peer.ID, location common.Location, data
 	// Check the received data type & hash matches the request
 	switch datatype.(type) {
 	case *types.Block:
-		if block, ok := recvdType.(*types.Block); ok && block.Hash() == data.(common.Hash) {
-			return block, nil
+		if block, ok := recvdType.(*types.Block); ok {
+			switch data := data.(type) {
+			case common.Hash:
+				if block.Hash() == data {
+					return block, nil
+				}
+				return nil, errors.Errorf("invalid response: expected block with hash %s, got %s", data, block.Hash())
+			case *big.Int:
+				nodeCtx := location.Context()
```

Comment:
one small nit-pick. This isn't the node's context. This is the request context. The node exists in all contexts now.